app/src/main/java/com/github/se/cyrcle/ui/map/MapScreen.kt

```diff
@@ -329,89 +325,46 @@ fun MapScreen(
                 mapView.gestures.addOnMoveListener(moveListener)
                 // ======================= MOVE LISTENER =======================
 
-                // ======================= MARKERS CLICK =======================
-                markerAnnotationManager?.addClickListener(
-                    OnPointAnnotationClickListener { pointAnnotation ->
-                      mapViewModel.removePreviewCard()
-                      // get the data from the PointAnnotation and deserialize it
-                      val parkingData = pointAnnotation.getData()?.asJsonObject
-                      val parkingDeserialized = gson.fromJson(parkingData, Parking::class.java)
-                      // move the camera to the selected parking
-                      mapViewportState.setCameraOptions {
-                        center(parkingDeserialized.location.center)
+                // ======================= ANNOTATION CLICK LISTENER =======================
+                // Define a common click handler function
+                fun handleAnnotationClick(
+                    annotation: Any,
```

Comment:
Here instead of using `Any` you could have used `Annotation` since it is the superclass of both PointAnnotation and PolygonAnnotation. This way we are more in control of what goes inside the function.

Reply:
I thought about that, but somehow it doesn't work. I get the following error: Incompatible types: PointAnnotation and Annotation. I don't really know how to do it

Reply:
The code is good as it is and the switch will handle this perfectly so it doesn't matter for this PR. Since you looked into i will not block the PR for it.